src/controllers/projectController.ts

```diff
@@ -50,14 +50,17 @@ export class ProjectController {
     }
 
     private async scaffoldJavaProject(basePath: string, projectName: string, javaVersion: number): Promise<boolean> {
-        const projectFile: string = path.join(basePath, projectName, ".project");
-        const extensionPath: string = this.context.extensionPath;
+        const projectRoot: string = path.join(basePath, projectName);
+        const templateRoot: string = path.join(this.context.extensionPath, "templates");
+        const projectFile: string = path.join(projectRoot, ".project");
         try {
+            fse.ensureDirSync(projectRoot);
```

Comment:
Can we just await fse.ensureDir?
A sync operation blocks others especially when there is potential retry logic.